solang-parser/src/solidity.lalrpop

```diff
@@ -752,8 +752,12 @@ NonIfStatement: Statement = {
     <l:@L> "unchecked" "{" <statements:Statement*> "}" <r:@R> => {
         Statement::Block { loc: Loc::File(file_no, l, r), unchecked: true, statements }
     },
-    <l:@L> "assembly" <assembly:AssemblyBlock> <r:@R> => {
-        Statement::Assembly { loc: Loc::File(file_no, l, r), assembly }
+    <l:@L> "assembly" <dialect:string?> <statements:AssemblyBlock> <r:@R> => {
```

Comment:
Here you us the string token, which will give the string, but not its location. If the contents is something we do not expect, then in sema we would like to give a warning with the correct location. Therefore it is better to `StringLiteral`.